File: objC-/objC-/ViewController.m
//
//  ViewController.m
//  objC-
//
//  Created by iPOS on 6/26/15.
//  Copyright (c) 2015 iPOS. All rights reserved.
//

#import "ViewController.h"

#define DarkSkyURL @"https://api.darkskyapp.com/v1/forecast/"
#define API_KEY @"a00788f170fa382851907a6f2752cda3"
#define LAT @"/42.7243"
#define LON @",-73.6927"

#define fullURL (DarkSkyURL API_KEY LAT LON)

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    dispatch_queue_t queue = dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0);
    dispatch_async(queue, ^{
        NSData *data = [NSData dataWithContentsOfURL:[NSURL URLWithString:fullURL]];
        [self performSelectorOnMainThread:@selector(fetchedForecast:) withObject:data waitUntilDone:YES];
    });
    
}

- (void)viewDidAppear:(BOOL)animated {
    
   
}

- (void)fetchedForecast:(NSData *) responeData {
    NSError *error;
    NSDictionary *json = [NSJSONSerialization JSONObjectWithData:responeData options: kNilOptions error:&error];
    NSLog(@"%@", json);
    self.lbTimeZone.text = [json objectForKey:@"timezone"];
    self.lbHourlyForecase.text = [json objectForKey:@"hourSummary"];
    self.lbTemperature.text = [NSString stringWithFormat:@"%@",[json objectForKey:@"currentTemp"]];
    
}
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

@end

File: iPOSOrder/iPOSOrder/ViewController.swift
//
//  ViewController.swift
//  iPOSOrder
//
//  Created by iPOS on 4/22/15.
//  Copyright (c) 2015 iPOS. All rights reserved.
//

import UIKit
import PKHUD
import Alamofire

class ViewController: UIViewController {
    
    let concurrentQueue : dispatch_queue_t = dispatch_queue_create("ipos.tuyen.pham.LoadingData", DISPATCH_QUEUE_CONCURRENT)
    let group = dispatch_group_create()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        PKHUD.sharedHUD.contentView = PKHUDSystemActivityIndicatorView()
        PKHUD.sharedHUD.show()
        
    }
    
    override func viewDidAppear(animated: Bool) {
      
     
        dispatch_group_notify(group, concurrentQueue, {
             //self.loadingListUser()
             self.loadingListArea()
             //self.loadingListFood()
        })


    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

   
    
    func loadingListFood() {
        Alamofire.request(.GET, "\(GlobalVariable.hostURL)/GetItems")
            .responseJSON { (request, response, json, error) in
                if(error != nil) {
                    NSLog("GetFood Error: \(error)")
                }
                else {
                    NSLog("GetFood Success! ")
                    var result = JSON(json!)
                    
                    //                    for (index: String, subJson: JSON) in jsonArray {
                    //
                    //                        var user = UserN()
                    //                        NSLog("User: \(subJson)")
                    //                        user.LoginName = subJson["LoginName"].stringValue
                    //                        user.UserName = subJson["UserName"].stringValue
                    //                        GlobalVariable.listUser.append(user)
                    //                    }
                    
                }
        }

        
    }
    
    func loadingListCombo() {
        
    }
    
    func loadingListUser() {
        Alamofire.request(.GET, "\(GlobalVariable.hostURL)/GetUsers")
            .responseJSON { (request, response, json, error) in
                if(error != nil) {
                    NSLog("GetUser Error: \(error)")
                }
                else {
                    NSLog("GetUser Success! ")
                    var result = JSON(json!)
                    NSLog("\(result) ")
//                    if let data = result["ListUser"].arrayValue as [JSON]?{
//                        NSLog("data \(data) ")
////                        for i in 0..< (data.count) {
////                            var user = UserN()
////                            NSLog("User: \(subJson)")
////                            user.LoginName = subJson["LoginName"].stringValue
////                            user.UserName = subJson["UserName"].stringValue
////                            GlobalVariable.listUser.append(user)
////                        }
//                    } else {
//                         NSLog("GetUser parse error! ")
//                    }

                    
                }
        }
    }
    
    func loadingListArea() {
        Alamofire.request(.GET, "\(GlobalVariable.hostURL)/GetAreas")
            .responseJSON { (request, response, json, error) in
                if(error != nil) {
                    NSLog("GetArea Error: \(error)")
                }
                else {
                    NSLog("GetArea Success! ")
                  
                    var result = JSON(json!)
                    
                    if let haystack = result["ListArea"].string {
                        
                            NSLog("Area: \(haystack)")
                        
                    }
                    NSLog("ListArea : \(result) ")

//                    NSLog("Area: \(json)  \(result)")
//                    var arrayArea = result[0]
//                    NSLog("Array Area: \(arrayArea.isArray)")
                    
                    //                    for (index: String, subJson: JSON) in jsonArray {
                    //
                    //                        var user = UserN()
                    //                        NSLog("User: \(subJson)")
                    //                        user.LoginName = subJson["LoginName"].stringValue
                    //                        user.UserName = subJson["UserName"].stringValue
                    //                        GlobalVariable.listUser.append(user)
                    //                    }
                    
                }
        }
    }

}

